### PR TITLE
feat(gatsby-source-graphql): upgrade apollo & graphql-tools deps

### PR DESCRIPTION
**Pull request by seanparmelee**
## Description

This is a continuation of #34077, where the goal was to bump `@graphql-tools/links` to pull in some newer transitive dependencies to ultimately fix a deprecation warning that shows up when using `gatsby-source-graphql` (for more context see #34042). As noted in the previous PR, `@apollo/client` is a peer dependency of `@graphql-tools/links` so that dependency was also brought in, which also replaces a number of deprecated apollo packages.

### Documentation

I updated all the docs & examples that referenced the older apollo packages.

## Related Issues

Fixes https://github.com/gatsbyjs/gatsby/issues/34042

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -28,6 +28,8 @@ packages/gatsby/gatsby-admin-public
 packages/gatsby-codemods/transforms
 packages/gatsby-source-graphql/batching
 packages/gatsby-plugin-gatsby-cloud/components
+packages/gatsby-plugin-gatsby-cloud/context
+packages/gatsby-plugin-gatsby-cloud/models
```
Comment (seanparmelee) [inline .eslintignore:32]: I was getting linting errors in these folders so added them to the ignore list as well

File: packages/gatsby-source-graphql/package.json [hunk 2]
```diff
@@ -7,12 +7,11 @@
...
+    "@apollo/client": "^3.5.8",
     "@babel/runtime": "^7.15.4",
-    "@graphql-tools/links": "^7.1.0",
-    "@graphql-tools/utils": "^7.10.0",
-    "@graphql-tools/wrap": "^7.0.8",
-    "apollo-link": "1.2.14",
-    "apollo-link-http": "^1.5.17",
+    "@graphql-tools/links": "^8.2.1",
+    "@graphql-tools/utils": "^8.6.1",
+    "@graphql-tools/wrap": "^8.3.3",
```
Comment (seanparmelee) [inline packages/gatsby-source-graphql/package.json:14]: since these `@graphql-tools` depend on each other, figured it made sense to bump all of them here

**Review by maglibula — APPROVED** [review]

**Comment by presto2116** [comment]
Is there any movement on this? it's been sitting for over a month now. The build log errors/warnings caused by gatsby-source-graphql are pretty irritating, especially since its a deep dependency error that has been fixed with this PR

**Comment by jamin84** [comment]
> Is there any movement on this? it's been sitting for over a month now. The build log errors/warnings caused by gatsby-source-graphql are pretty irritating, especially since its a deep dependency error that has been fixed with this PR

Also looking forward to getting this merged!

**Review by wardpeet — APPROVED** [review]
I couldn't find anything breaking in dependencies so let's merge and get some feedback on our next channel. 

Thank you for upgrading!

**Pull request merged**